src/lib/GenericPredictor.cc

```diff
@@ -115,6 +127,20 @@ void GenericPredictor::flush(uint64_t address) {
     }
     rasHistory_.erase(it);
   }
+
+  // If possible, pop instruction from FTQ
+  FTQ_.pop_back();
+
+  // Roll back global history
+  globalHistory_ >>= 1;
```

Comment:
If a flush occurs, do we know it will always be because of the most recently processed branch?